util/src/main/java/com/coderanch/util/cli/InputUtility.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (C) 2018 Coderanch.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.coderanch.util.cli;
+
+import java.io.*;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * Multipurpose input utility for use with a command line interface.
+ * The user can pass a prompt and optional validation.
+ * The utility will read the provided input stream and return a possibly validated String or primitive.
+ */
+public final class InputUtility implements Closeable, AutoCloseable {
+
+    /**
+     * Possible inputs for yes.
```

Comment:
I think the following reads a bit more clearly:
```suggestion
     * Allowed user inputs that are considered synonymous with "yes".
```